File: smart pay app/app/src/main/java/com/smartpay/application/BaseActivity.java
package com.smartpay.application;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;
import android.widget.Toolbar;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;

abstract class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void setToolbar(String title, Boolean isBackVisible){
        androidx.appcompat.widget.Toolbar mToolbar = findViewById(R.id.tb_header);
        setSupportActionBar(mToolbar);
        ActionBar actionBar = getSupportActionBar();
        if(actionBar != null){
            actionBar.setTitle(null);
            actionBar.setDisplayHomeAsUpEnabled(isBackVisible);
        }
        mToolbar.setVisibility(View.VISIBLE);
        TextView tvHeader = findViewById(R.id.tv_header);
        tvHeader.setText(title);
    }

//    public void setBackIcon(){
//        supportActionBar?.setDisplayHomeAsUpEnabled(true)
//        toolbar.setNavigationOnClickListener {
//            finish()
//        }
//        ActionBar actionBar = getSupportActionBar();
//        if(actionBar != null) {
//            actionBar.setDisplayHomeAsUpEnabled(true);
//            actionBar.
//        }
//
//    }
}

File: smart pay app/app/src/main/java/com/smartpay/application/BaseActivity.kt
package com.smartpay.application

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import com.smartpay.application.R
import android.widget.TextView
import androidx.appcompat.widget.Toolbar

abstract class BaseActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun setToolbar(title: String?, isBackVisible: Boolean) {
        val mToolbar = findViewById<Toolbar>(R.id.tb_header)
        setSupportActionBar(mToolbar)
        supportActionBar?.title = null
        supportActionBar?.setDisplayHomeAsUpEnabled(isBackVisible)
        mToolbar.visibility = View.VISIBLE
        val tvHeader = findViewById<TextView>(R.id.tv_header)
        tvHeader.text = title
    }
}